ceph-dashboard-cephadm-e2e-nightly/config/definitions/ceph-dashboard-cephadm-e2e-nightly.yml

```
@@ -71,6 +71,6 @@
       - email:
           recipients: ceph-qa@ceph.io
       - archive:
-          artifacts: 'logs/*.log'
+          artifacts: 'logs/**'
```

I don't get a good explanation for that behaviour, but specifying the extension is a good practice to avoid unintentionally archiving temporary files or other random stuff:
```suggestion
          artifacts: 'logs/**/*.log'
```

---

@epuertat Doing a pattern like this would create a dirty looking log like this in the[ job](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/3353/consoleFull)
```
Archiving artifacts
hudson.FilePath$ValidateAntFileMask$1Cancel
	at hudson.FilePath$ValidateAntFileMask$1.isCaseSensitive(FilePath.java:3209)
	at org.apache.tools.ant.DirectoryScanner.lambda$isIncluded$3(DirectoryScanner.java:1374)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:528)
	at org.apache.tools.ant.DirectoryScanner.isIncluded(DirectoryScanner.java:1374)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1241)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1267)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1267)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1267)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1267)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1267)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1267)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1267)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1267)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1267)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1267)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1194)
	at org.apache.tools.ant.DirectoryScanner.scandir(DirectoryScanner.java:1156)
	at org.apache.tools.ant.DirectoryScanner.checkIncludePatterns(DirectoryScanner.java:954)
	at org.apache.tools.ant.DirectoryScanner.scan(DirectoryScanner.java:912)
	at hudson.FilePath$ValidateAntFileMask.hasMatch(FilePath.java:3222)
Caused: java.lang.InterruptedException: no matches found within 10000
	at hudson.FilePath$ValidateAntFileMask.hasMatch(FilePath.java:3227)
	at hudson.FilePath$ValidateAntFileMask.invoke(FilePath.java:3135)
	at hudson.FilePath$ValidateAntFileMask.invoke(FilePath.java:3083)
	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:3487)
Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to JNLP4-connect connection from 8.43.84.3/8.43.84.3:46074
		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1785)
		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:356)
		at hudson.remoting.Channel.call(Channel.java:1000)
		at hudson.FilePath.act(FilePath.java:1194)
		at hudson.FilePath.act(FilePath.java:1183)
		at hudson.FilePath.validateAntFileMask(FilePath.java:3080)
		at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:269)
		at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:79)
		at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
		at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:814)
		at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:763)
		at hudson.model.Build$BuildExecution.post2(Build.java:179)
		at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:707)
		at hudson.model.Run.execute(Run.java:1921)
		at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:44)
		at hudson.model.ResourceController.execute(ResourceController.java:101)
		at hudson.model.Executor.run(Executor.java:442)
Caused: hudson.FilePath$TunneledInterruptedException
	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:3489)
	at hudson.remoting.UserRequest.perform(UserRequest.java:211)
	at hudson.remoting.UserRequest.perform(UserRequest.java:54)
	at hudson.remoting.Request$2.run(Request.java:376)
	at hudson.remoting.InterceptingExecutorService.lambda$wrap$0(InterceptingExecutorService.java:78)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at hudson.remoting.Engine$1.lambda$newThread$0(Engine.java:121)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused: java.lang.InterruptedException: java.lang.InterruptedException: no matches found within 10000
	at hudson.FilePath.act(FilePath.java:1196)
	at hudson.FilePath.act(FilePath.java:1183)
	at hudson.FilePath.validateAntFileMask(FilePath.java:3080)
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:269)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:79)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:814)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:763)
	at hudson.model.Build$BuildExecution.post2(Build.java:179)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:707)
	at hudson.model.Run.execute(Run.java:1921)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:44)
	at hudson.model.ResourceController.execute(ResourceController.java:101)
	at hudson.model.Executor.run(Executor.java:442)
No artifacts found that match the file pattern "logs/**/*.log". Configuration error?
```
as opposed to the one with the [changes in this PR](https://jenkins.ceph.com/job/ceph-dashboard-cephadm-e2e/3348/console).
```
Archiving artifacts
No artifacts found that match the file pattern "logs/**". Configuration error?
```

---

@djgalloway As you can[ see here](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh#L25), when there is an error I create a folder called the log inside the workspace and I save the logs in that directory which is then used to archive and compress. So for a successful run this logs folder won't be created and that's where I think the pattern fails.

---

Yeah, so I'm just suggesting move that `mv` out of the `on_error` function to a spot in the script where it will always get run.

---

Let's merge it as it is right now!

---

Oh, geez, sorry. I misunderstood and thought this change as-is would be a breaking change.